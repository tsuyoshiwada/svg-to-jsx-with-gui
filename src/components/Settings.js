import React, { Component } from 'react';
import TransitionGroup from 'react-transition-group/TransitionGroup';
import styled from 'styled-components';
import SettingsDialog from './SettingsDialog';
import Button from './internal/Button';
import { Cog } from '../icons/';
import firstChild from '../utils/firstChild';


const OpenButton = Button.extend`
  position: fixed;
  right: 20px;
  bottom: 80px;
  z-index: 999;
  height: 50px;
  padding: 0 30px;
  background: #353954;
  border: 3px solid #fff;
  border-radius: 50px;
  box-shadow: none;
  color: #fff;
  font-weight: bold;

  & svg {
    width: 1em;
    height: 1em;
    margin: 0 0.5em 0 0;
    vertical-align: middle;
  }

  &:hover {
    background-color: #fff;
    color: #4c5275;
    box-shadow: 0 2px 3px rgba(0, 0, 0, 0.3);
  }

  &:active {
    background-color: #eaeaea;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.11) inset;
  }
`;


export default class Settings extends Component {
  state = {
    open: false,
  };

  handleOpenClick = (e) => {
    e.preventDefault();
    this.setState({ open: true });
  };

  handleClose = () => {
    this.setState({ open: false });
  };

  render() {
    const {
      open,
    } = this.state;

    return (
      <div>
        {/* Dialog */}
        <TransitionGroup component={firstChild}>
          {open &&
            <SettingsDialog
              onRequestClose={this.handleClose}
            />
          }
        </TransitionGroup>

        {/* Open */}
        <OpenButton
          onClick={this.handleOpenClick}
        >
          <Cog />
          Settings
        </OpenButton>
      </div>
    );
  }
}
