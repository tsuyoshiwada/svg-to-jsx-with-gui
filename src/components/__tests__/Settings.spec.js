import React from 'react';
import { shallow } from 'enzyme';
import Settings, { OpenButton } from '../Settings';


describe('components#<Settings />', () => {
  test('Should be render', () => {
    const wrapper = shallow(<Settings />);
    expect(wrapper).toMatchSnapshot();
  });


  test('Should be render when open', () => {
    const wrapper = shallow(<Settings />);
    wrapper.setState({ ...wrapper.state(), open: true });
    expect(wrapper).toMatchSnapshot();
  });


  test('Should be dialog open when <OpenButton /> click', () => {
    const wrapper = shallow(<Settings />);

    expect(wrapper.state('open')).toBe(false);

    wrapper.find(OpenButton).simulate('click', { preventDefault: () => null });
    expect(wrapper.state('open')).toBe(true);
  });
});
