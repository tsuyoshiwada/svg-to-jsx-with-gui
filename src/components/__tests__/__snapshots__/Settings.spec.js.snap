// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components#<Settings /> Should be render 1`] = `
<div>
  <CSSTransitionGroup
    component={[Function]}
    transitionAppear={false}
    transitionEnter={true}
    transitionEnterTimeout={320}
    transitionLeave={true}
    transitionLeaveTimeout={120}
    transitionName="dialog"
  />
  <styled.button
    onClick={[Function]}
  >
    <Cog />
    <span>
      Settings
    </span>
  </styled.button>
</div>
`;

exports[`components#<Settings /> Should be render when open 1`] = `
<div>
  <CSSTransitionGroup
    component={[Function]}
    transitionAppear={false}
    transitionEnter={true}
    transitionEnterTimeout={320}
    transitionLeave={true}
    transitionLeaveTimeout={120}
    transitionName="dialog"
  >
    <SettingsDialog
      onRequestClose={[Function]}
    />
  </CSSTransitionGroup>
  <styled.button
    onClick={[Function]}
  >
    <Cog />
    <span>
      Settings
    </span>
  </styled.button>
</div>
`;
